### DISCUSSION TITLE
.NET Maui w/ ASP.NET Core backend, Web Authenticator integration help

### DISCUSSION THREAD

**Discussion by harrismasterson** [dotnet/maui#12802] [Q&A]
I'm working on .NET Maui mobile project, and I'm trying to achieve the look and the UX in the image below (i.e., I do NOT want to use MSAL).

I've created [ASP.NET](https://asp.net/) core server backend ([ASP.NET](https://asp.net/) Core API project template) as described on [Web Authenticator](https://learn.microsoft.com/en-us/dotnet/maui/platform-integration/communication/authentication?view=net-maui-7.0&tabs=ios) page, and have followed the steps to add authentication service to this backend project + configured Google authentication in the project w/ client id and client secrets as described [here](https://learn.microsoft.com/en-us/aspnet/core/security/authentication/social/?tabs=visual-studio&view=aspnetcore-7.0https://learn.microsoft.com/en-us/aspnet/core/security/authentication/social/?tabs=visual-studio&view=aspnetcore-7.0) and [here](https://learn.microsoft.com/en-us/aspnet/core/security/authentication/social/google-logins?view=aspnetcore-7.0).

Note - I created an additional [ASP.NET](https://asp.net/) Core Web app project, and verified that the login with google functionality works w/ the client id and client secrets I've generated

The ASP.NET Core project runs w/o errors and renders a simple index page in browser via controller set up in project, and I've also added a mobile auth controller based on [this repo](https://github.com/dotnet/maui/blob/main/src/Essentials/samples/Sample.Server.WebAuthenticator/Controllers/MobileAuthController.cs). What I'm now stuck on is trying to connect my backend with my .NET maui framework so that I can have a page such as the one below, and then the buttons on the screen link to authentication service I've built out in backend. Any suggestions/resources/guidance is appreciated, this is my first mobile dev project + my first project in general where I have to start w/ nothing.
![Screenshot 2023-01-18 at 1 36 03 PM](https://user-images.githubusercontent.com/74700718/213581574-91f36763-b15c-4a96-bfa8-b594067ed470.png)

**Comment by hiyelbaz** [comment]
Hi, @harrismasterson did you find a solution?

**Comment by harrismasterson** [comment]
Have not found a totally satisfying solution for how to handle Authentication in general using .NET Maui - seems that there is not yet a one size fits all solution that allows me to achieve the above UX. Please let me know if you have found anything
